db/view/view.cc

```diff
@@ -1948,21 +1948,17 @@ future<> view_builder::start(service::migration_manager& mm) {
 
 future<> view_builder::drain() {
     if (_as.abort_requested()) {
-        return make_ready_future();
+        co_return;
     }
     vlogger.info("Draining view builder");
     _as.request_abort();
-    return _started.then([this] {
-        return _mnotifier.unregister_listener(this).then([this] {
-            return _sem.wait();
-        }).then([this] {
-            _sem.broken();
-            return _build_step.join();
-        }).finally([this] {
-            return parallel_for_each(_base_to_build_step, [] (std::pair<const table_id, build_step>& p) {
-                return p.second.reader.close();
-            });
-        });
+    co_await std::move(_started);
```

Comment:
If this throws, the whole stop sequence gets aborted, because .stop() is called from inside deferred action in main, which is, in turn, is either due to handling another exception, or regular stop. In any case it bails out and just exits

Reply:
Or even better -- if this throws, then steps list is empty and it makes no sense in closing its readers

Reply:
OK, makes sense.